protocols/mdns/src/behaviour/iface.rs

```diff
@@ -198,43 +199,51 @@ impl InterfaceState {
         params: &impl PollParameters,
     ) -> Option<(PeerId, Multiaddr, Instant)> {
         // Poll receive socket.
-        while self.recv_socket.poll_readable(cx).is_ready() {
-            match self
-                .recv_socket
-                .recv_from(&mut self.recv_buffer)
-                .now_or_never()
-            {
-                Some(Ok((len, from))) => {
+        match self.recv_socket.poll_read(cx, &mut self.recv_buffer) {
```

Comment:
This should be a loop similar as to before.

Previously, we read as many packets as possible until we hit `Poll::Pending`. Now we are reading at most one packet and then moving on to sending responses. We should exhaust the `recv_socket` until it returns `Pending`.